packages/astro/src/prefetch/index.ts

```diff
@@ -241,6 +242,12 @@ function elMatchesStrategy(el: EventTarget | null, strategy: string): el is HTML
 	if (attrValue === 'false') {
 		return false;
 	}
+
+	// Fallback to tap strategy if using data saver mode or slow connection
+	if ((attrValue != null || prefetchAll) && isSlowConnection()) {
+		return strategy === 'tap';
+	}
```

Comment:
I think this check could indirectly affect hover and viewport strategies hooking its initial listeners.

Some events like `mouseenter` and `mouseleave`, and `IntersectionObserver` can only be initialized on startup, which calls this function. And if on startup we found a slow connection, we will want to avoid this check because hover/viewport strategies could continue to work when they later have a stable connection.

I think the fix here is something like this:
```suggestion
	// Fallback to tap strategy if using data saver mode or slow connection
	if (strategy === 'tap' && (attrValue != null || prefetchAll) && isSlowConnection()) {
		return true;
	}
```